ui/apps/platform/src/Containers/Collections/CollectionsTablePage.tsx

```diff
@@ -31,17 +34,30 @@ const sortOptions = {
 };
 
 function CollectionsTablePage({ hasWriteAccessForCollections }: CollectionsTablePageProps) {
+    const { searchFilter, setSearchFilter } = useURLSearch();
+    const pagination = useURLPagination(20);
+    const { page, perPage, setPage } = pagination;
     const { sortOption, getSortParams } = useURLSort(sortOptions);
-    const listQuery = useCallback(() => listCollections({}, sortOption, 0, 20), [sortOption]);
+
+    const listQuery = useCallback(
+        () => listCollections(searchFilter, sortOption, page - 1, perPage),
+        [searchFilter, sortOption, page, perPage]
+    );
     const { data: listData, loading: listLoading, error: listError } = useRestQuery(listQuery);
 
-    const countQuery = useCallback(() => getCollectionCount({}), []);
+    const countQuery = useCallback(() => getCollectionCount(searchFilter), [searchFilter]);
     const { data: countData, loading: countLoading, error: countError } = useRestQuery(countQuery);
-
     const isDataAvailable = typeof listData !== 'undefined' && typeof countData !== 'undefined';
     const isLoading = !isDataAvailable && (listLoading || countLoading);
     const loadError = listError || countError;
 
+    useEffectAfterFirstRender(() => {
+        // Prevent viewing a page beyond the maximum page count
+        if (typeof countData !== 'undefined' && page > Math.ceil(countData / perPage)) {
```

Comment:
It would be nice to roll this logic into the `useURLPagination` hook if possible. Maybe as an additional follow up.

Reply:
Yes, `useEffectAfterFirstRender` hooks feels awkward.

Reply:
Forgive me for asking instead of studying. Does change to search filter reset to page 1?

Reply:
It does not, but I think maybe it should. As for prior art:

- Violations _does_ reset page when the search filter changes
- Risk Acceptance _does not_ reset page when the search filter changes
- Policies doesn't support server side pagination

Let me take another look at this and see if it is worth changing to remove some of the awkwardness in this PR, or if I can come up with a good follow up plan.

Reply:
I've got a simple alternate approach to this that I'll push up instead that relies on https://github.com/stackrox/stackrox/pull/3228 to function properly.

Essentially just calling `setPage(1); setSearchFilter(value);` when the filter changes, but requires a fix to the `useURLParameter()` hook to work.